File: Java/프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.java
class Solution {
    public static String solution(String my_string, String overwrite_string, int s) {
        String answer = "";

        for (int i = 0; i < my_string.length(); i++) {
            if (i < s) {
                answer += my_string.charAt(i);
            } else if (i == s) {
                answer += overwrite_string;
                i += overwrite_string.length() - 1;
            } else if (i > s) {
                answer += my_string.charAt(i);
            }
        }
        return answer;
    }
}

File: JavaScript/프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.js
function solution(my_string, overwrite_string, s) {
    let answer = [...my_string];
    answer.splice(s, overwrite_string.length, overwrite_string);
    // splice(startIndex, deleteLength, changeItem) 순으로 작성
    // startIndex에서 deleteLength만큼을 changeItem으로 대체한다 
    return answer.join('');
}